File: python/LeetCode/01437.py
from typing import List

def kLengthApart(self, nums: List[int], k: int) -> bool:
    ones = []
    for i in range(len(nums)):
        if nums[i] == 1:
            ones.append(i)
    
    if len(ones) <= 1:
        return True

    for i in range(1, len(ones)):
        if ones[i] - ones[i-1] <= k:
            return False
    

    return True

File: go/leetcode/01437.go
package main

func kLengthApart(nums []int, k int) bool {
	n := len(nums)
	ones := make([]int, 0)

	for i := range n {
		if nums[i] == 1 {
			ones = append(ones, i)
		}
	}

	j := len(ones)
	if j <= 1 {
		return true
	}

	for i := range j - 1 {
		if ones[i+1]-ones[i] < k+1 {
			return false
		}
	}

	return true
}
